envoy/ssl/context_manager.h

```diff
@@ -30,7 +36,8 @@ class ContextManager {
    */
   virtual ServerContextSharedPtr
   createSslServerContext(Stats::Scope& scope, const ServerContextConfig& config,
-                         const std::vector<std::string>& server_names) PURE;
+                         const std::vector<std::string>& server_names,
+                         ContextAdditionalInitFunc additional_init) PURE;
```

Comment:
Instead of passing in a nullable function pointer, why not just pass in a bool and initialize quic cert per string and private key in ServerContextImpl constructor?

Reply:
I was trying to keep as much quic-specific code in `source/common/quic` is possible.